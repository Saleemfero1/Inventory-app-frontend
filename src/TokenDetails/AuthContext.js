import React, { createContext, useState } from "react";

export const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
  const [isAuthenticated, setIsAuthenticated] = useState(true);
  const [token, setToken] = useState(sessionStorage.getItem("token"));
  const [userName, setUser] = useState(sessionStorage.getItem("username"));
  const [dashData, setDashData] = useState([]);
  const [itemData, setItemData] = useState([]);
  const [locationData, setLocationData] = useState([]);
  const [supplyData, setSupplyData] = useState([]);
  const [demandData, setDemandData] = useState([]);
  const [thresholdData, setThresholdData] = useState([]);
  const [totalItem, setTotalItem] = useState(0);
  const [totalActiveItem, setTotalActiveItem] = useState(0);
  const [totalSupply, setTotalSupply] = useState(0);
  const [totalDemand, setTotalDemand] = useState(0);
  const [digg, setDigg] = useState({});
  const [totalLocation, setTotalLocation] = useState(0);
  const [TransactionData, setTransactionData] = useState([]);
  const [lowAndHighStock, setlowAndHighStock] = useState([]);
  const [TrendingItems, SetTrendingItems] = useState([]);
  const updateToken = (newToken, newUser) => {
    if (newToken) {
      setToken(newToken);
    }
    setUser(newUser);
  };

  return (
    <AuthContext.Provider
      value={{
        token,
        updateToken,
        userName,
        dashData,
        setDashData,
        setItemData,
        itemData,
        locationData,
        setLocationData,
        supplyData,
        setSupplyData,
        demandData,
        setDemandData,
        thresholdData,
        setThresholdData,
        isAuthenticated,
        setIsAuthenticated,
        totalItem,
        setTotalItem,
        totalActiveItem,
        setTotalActiveItem,
        setTotalSupply,
        totalSupply,
        setTotalDemand,
        totalDemand,
        totalLocation,
        setTotalLocation,
        digg,
        setDigg,
        TransactionData,
        setTransactionData,
        lowAndHighStock,
        setlowAndHighStock,
        TrendingItems,
        SetTrendingItems,
      }}
    >
      {children}
    </AuthContext.Provider>
  );
};
