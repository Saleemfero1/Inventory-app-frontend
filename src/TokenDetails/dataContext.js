// import React, { createContext, useState } from "react";

// export const DataContext = createContext();

// export const DataProvider = ({ children }) => {
//   const [dashData, setData] = useState([]);

//   const updateData = (newData) => {
//     setData(newData);
//   };

//   return (
//     <DataContext.Provider value={{ dashData, updateData }}>
//       {children}
//     </DataContext.Provider>
//   );
// };
