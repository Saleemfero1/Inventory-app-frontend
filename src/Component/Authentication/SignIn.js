import * as React from "react";
import Avatar from "@mui/material/Avatar";
import Button from "@mui/material/Button";
import CssBaseline from "@mui/material/CssBaseline";
import TextField from "@mui/material/TextField";
import FormControlLabel from "@mui/material/FormControlLabel";
import Checkbox from "@mui/material/Checkbox";
import Link from "@mui/material/Link";
import Grid from "@mui/material/Grid";
import Box from "@mui/material/Box";
import LockOutlinedIcon from "@mui/icons-material/LockOutlined";
import Typography from "@mui/material/Typography";
import Container from "@mui/material/Container";
import { createTheme, ThemeProvider } from "@mui/material/styles";
import { useState } from "react";
import AuthServices from "../../Service/AuthServices";
import { toast, ToastContainer } from "react-toastify";
import { useNavigate } from "react-router-dom";
import { AuthContext } from "../../TokenDetails/AuthContext";
import { useEffect } from "react";

function Copyright(props) {
  return (
    <Typography variant="body2" color="#6a7ae4" align="center" {...props}>
      {"Copyright Â© "}
      <Link color="inherit" href="https://mui.com/">
        TupleInventory.com
      </Link>{" "}
      {new Date().getFullYear()}
      {"."}
    </Typography>
  );
}

const defaultTheme = createTheme();

export default function SignIn() {
  const { updateToken } = React.useContext(AuthContext);
  const navigate = useNavigate();
  const [user, setUser] = useState({
    username: "",
    password: "",
  });

  const handleSubmit = (event) => {
    event.preventDefault();

    AuthServices.logInuser(user)
      .then((response) => {
        sessionStorage.setItem("token", response.data.accessToken);
        sessionStorage.setItem("username", user.username);
        updateToken(response.data.accessToken, user.username);
        navigate("/dashboard");
      })
      .catch((err) => {
        toast.error("unauthorized user");
        console.log(err);
      });
  };

  const onChangeUserName = (event) => {
    setUser({ ...user, username: event.target.value });
  };

  const onChangePassword = (event) => {
    setUser({ ...user, password: event.target.value });
  };

  return (
    <ThemeProvider theme={defaultTheme}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <Box
          sx={{
            marginTop: 8,
            display: "flex",
            flexDirection: "column",
            alignItems: "center",
          }}
        >
          <Avatar sx={{ m: 1, bgcolor: "#6a7ae4" }}>
            <LockOutlinedIcon />
          </Avatar>
          <Typography component="h1" variant="h5">
            Sign in
          </Typography>
          <Box
            component="form"
            onSubmit={handleSubmit}
            noValidate
            sx={{ mt: 1 }}
          >
            <TextField
              margin="normal"
              required
              fullWidth
              id="username"
              label="username"
              name="username"
              value={user.username}
              onChange={onChangeUserName}
              autoComplete="username"
              autoFocus
            />
            <TextField
              margin="normal"
              required
              fullWidth
              name="password"
              label="Password"
              type="password"
              id="password"
              value={user.password}
              onChange={onChangePassword}
              autoComplete="current-password"
            />
            <FormControlLabel
              control={<Checkbox value="remember" color="primary" />}
              label="Remember me"
            />
            <Button
              type="submit"
              fullWidth
              variant="contained"
              sx={{ mt: 3, mb: 2, bgcolor: "#6a7ae4" }}
            >
              Sign In
            </Button>
            <Grid container>
              <Grid item xs>
                <Link href="#" variant="body2" color={"#6a7ae4"}>
                  Forgot password?
                </Link>
              </Grid>
              <Grid item>
                <Link href="/register" variant="body2" color={"#6a7ae4"}>
                  {"Don't have an account? Sign Up"}
                </Link>
              </Grid>
            </Grid>
          </Box>
        </Box>
        <Copyright sx={{ mt: 8, mb: 4 }} />
      </Container>
    </ThemeProvider>
  );
}
