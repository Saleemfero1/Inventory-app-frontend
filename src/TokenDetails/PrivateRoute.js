import React, { useContext, useState } from "react";
import { Navigate } from "react-router-dom";
import { AuthContext } from "./AuthContext";

const PrivateRoute = ({ children }) => {
  const { token } = useContext(AuthContext);
  return token ? <>{children}</> : <Navigate to="/" />;
};

export default PrivateRoute;
