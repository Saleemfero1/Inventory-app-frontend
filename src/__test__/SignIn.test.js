import renderer from "react-test-renderer";
import { AuthProvider } from "../TokenDetails/AuthContext";
import { MemoryRouter } from "react-router-dom";
import { render, fireEvent, waitFor, screen } from "@testing-library/react";
import SignIn from "../Component/Authentication/SignIn";
it("renders correctly", () => {
  const tree = renderer
    .create(
      <AuthProvider>
        <MemoryRouter>
          <SignIn />
        </MemoryRouter>
      </AuthProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();

  const treeOne = renderer
    .create(
      <AuthProvider>
        <MemoryRouter>
          <SignIn></SignIn>
        </MemoryRouter>
      </AuthProvider>
    )
    .toJSON();
  expect(treeOne).toMatchSnapshot();
});

it("renders correctly", async () => {
  render(
    <AuthProvider>
      <MemoryRouter>
        <SignIn />
      </MemoryRouter>
    </AuthProvider>
  );
  expect(screen.getByTestId("LockOutlinedIcon")).toBeInTheDocument();
  expect(screen.getByRole("heading", { name: /sign in/i }));
  expect(
    screen.getByRole("textbox", { name: /username/i })
  ).toBeInTheDocument();
  expect(screen.getByLabelText(/password/i)).toBeInTheDocument();
  expect(screen.getByRole("button", { name: /sign in/i })).toBeInTheDocument();
  expect(
    screen.getByRole("link", { name: /don't have an account\? sign up/i })
  ).toBeInTheDocument();
  expect(screen.getByText(/copyright Â© 2023\./i)).toBeInTheDocument();
  expect(
    screen.getByRole("link", { name: /tupleinventory\.com/i })
  ).toBeInTheDocument();
});

describe("SignIn", () => {
  test("should show alert when username is not entered", async () => {
    const { getByRole, getByText } = render(
      <AuthProvider>
        <MemoryRouter>
          <SignIn />
        </MemoryRouter>
      </AuthProvider>
    );
    const submitButton = getByRole("button", { name: "Sign In" });

    fireEvent.click(submitButton);

    await waitFor(() => {
      const alertMessage = getByText("Please enter your username.");
      expect(alertMessage).toBeInTheDocument();
    });
  });
});
