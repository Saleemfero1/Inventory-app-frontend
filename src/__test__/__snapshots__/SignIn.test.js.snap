// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="div-body"
>
  <div
    className="Toastify"
  />
  <div
    className="div-justify"
  >
    <main
      className="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
    >
      <div
        className="MuiBox-root css-63otnq"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1xvo4iu-MuiAvatar-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LockOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <h1
          className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          data-testid="Signin"
        >
          Sign in
        </h1>
        <form
          className="MuiBox-root css-164r41r"
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
            data-testid="a1"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="username"
              id="username-label"
            >
              username
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="username"
                autoFocus={true}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="username"
                name="username"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="enter user name"
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    username
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="Password"
              id="Password-label"
            >
              Password
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="current-password"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="Password"
                name="Password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Password"
                required={false}
                type="Password"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-hx3d2o-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            Sign In
          </button>
          <div
            className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <a
                className="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-1cnytyu-MuiTypography-root-MuiLink-root"
                href="/signup"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                Don't have an account? Sign Up
              </a>
            </div>
          </div>
        </form>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-18r3eod-MuiTypography-root"
      >
        Copyright Â© 
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-duj00d-MuiTypography-root-MuiLink-root"
          href="https://mui.com/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          TupleInventory.com
        </a>
         
        2023
        .
      </p>
    </main>
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="div-body"
>
  <div
    className="Toastify"
  />
  <div
    className="div-justify"
  >
    <main
      className="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
    >
      <div
        className="MuiBox-root css-63otnq"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1xvo4iu-MuiAvatar-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LockOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <h1
          className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          data-testid="Signin"
        >
          Sign in
        </h1>
        <form
          className="MuiBox-root css-164r41r"
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
            data-testid="a1"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="username"
              id="username-label"
            >
              username
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="username"
                autoFocus={true}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="username"
                name="username"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="enter user name"
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    username
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="Password"
              id="Password-label"
            >
              Password
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="current-password"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="Password"
                name="Password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Password"
                required={false}
                type="Password"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-hx3d2o-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            Sign In
            <span
              className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <a
                className="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-1cnytyu-MuiTypography-root-MuiLink-root"
                href="/signup"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                Don't have an account? Sign Up
              </a>
            </div>
          </div>
        </form>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-18r3eod-MuiTypography-root"
      >
        Copyright Â© 
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-duj00d-MuiTypography-root-MuiLink-root"
          href="https://mui.com/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          TupleInventory.com
        </a>
         
        2023
        .
      </p>
    </main>
  </div>
</div>
`;
